eval/lib/saml.go

```diff
@@ -19,11 +17,24 @@ const (
 	NameIdFormatUnspecified = "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified"
 )
 
-func NewSamlSsoUrlFunction(samlConfigs []*config.SAML) function.Function {
-	samls := make(map[string]*config.SAML)
-	for _, s := range samlConfigs {
-		samls[s.Name] = s
+func NewSamlSsoUrlFunction(configs []*config.SAML) function.Function {
+	type entity struct {
+		config     *config.SAML
+		descriptor *types.EntityDescriptor
+		err        error
 	}
+
+	samlEntities := make(map[string]*entity)
+	for _, conf := range configs {
+		metadata := &types.EntityDescriptor{}
+		err := xml.Unmarshal(conf.MetadataBytes, metadata)
```

Comment:
Can we have also the JSON parsing in configload in order not having to parse it twice (function and access control)?

Reply:
This is part of this explicit implementation because I wouldn't add related types (types.EntityDescriptor{}) to more generic configload package. Let's improve this in a second run maybe after #247 has been merged.

Reply:
ok